core/templates/dev/head/domain/exploration/QuestionObjectFactory.js

```diff
@@ -80,8 +80,8 @@ oppia.factory('QuestionObjectFactory', [function() {
     return this._answerGroups.length > 0;
   };
 
-  Question.prototype.getId = function () {
-    return this.getStateName().toLowerCase().split(' ').join('');
+  Question.prototype.getId = function() {
```

Comment:
Add a comment here to say that the IDs generated by this method should not contain SIDEBAR_ID_DELIMITER, in order to prevent interference with the sidebar IDs.